node/src/main/scala/coop/rchain/node/web/Transaction.scala

```diff
@@ -0,0 +1,257 @@
+      deriveEncoder[TransactionResponse]
+
+    implicit val decodeTransaction: Decoder[Transaction]         = deriveDecoder[Transaction]
+    implicit val decodeTransactionType: Decoder[TransactionType] = deriveDecoder[TransactionType]
+    implicit val decodeTransactionInfo: Decoder[TransactionInfo] = deriveDecoder[TransactionInfo]
+    implicit val decodeTransactionResponse: Decoder[TransactionResponse] =
+      deriveDecoder[TransactionResponse]
+  }
+
+  object SCodec {
+    import scodec._
+    import scodec.bits._
+    import scodec.codecs._
+    import coop.rchain.rholang.interpreter.storage._
+
+    val transactionCodec: Codec[Transaction] =
+      (utf8_32L :: utf8_32L :: long(63) :: serializePar.toSizeHeadCodec :: optional[String](
+        bool,
+        utf8_32L
+      )).as[Transaction]
+    val precharge: Codec[PreCharge]   = utf8_32L.as[PreCharge]
+    val refund: Codec[Refund]         = utf8_32L.as[Refund]
+    val user: Codec[UserDeploy]       = utf8_32L.as[UserDeploy]
+    val closeBlock: Codec[CloseBlock] = utf8_32L.as[CloseBlock]
+    val slash: Codec[SlashingDeploy]  = utf8_32L.as[SlashingDeploy]
+    val transactionType: Codec[TransactionType] = discriminated[TransactionType]
+      .by(uint8)
+      .subcaseP(0) {
+        case e: PreCharge => e
+      }(precharge)
+      .subcaseP(1) {
+        case s: UserDeploy => s
+      }(user)
+      .subcaseP(2) {
+        case pb: Refund => pb
+      }(refund)
+      .subcaseP(3) {
+        case c: CloseBlock => c
+      }(closeBlock)
+      .subcaseP(4) {
+        case s: SlashingDeploy => s
+      }(slash)
+
+    val transactionInfo: Codec[TransactionInfo] =
+      (transactionCodec :: transactionType).as[TransactionInfo]
+    val transactionResponseCodec: Codec[TransactionResponse] = list(transactionInfo)
+      .as[TransactionResponse]
+  }
+
+  // This is the hard-coded unforgeable name for
+  // https://github.com/rchain/rchain/blob/43257ddb7b2b53cffb59a5fe1d4c8296c18b8292/casper/src/main/resources/RevVault.rho#L25
+  // This hard-coded value is only useful with current(above link version) `RevVault.rho` implementation but it is
+  // useful for all the networks(testnet, custom network and mainnet) starting with this `RevVault.rho`.
+  //
+  // This hard-coded value needs to be changed when
+  // 1. `RevVault.rho` is changed
+  // 2. [[coop.rchain.casper.genesis.contracts.StandardDeploys.revVault]] is changed
+  // 3. The random seed algorithm for unforgeable name of the deploy is changed
+  //
+  // This is not needed when onChain transfer history is implemented and deployed to new network in the future.
+  val transferUnforgeable = {
+    val seedForRevVault = Tools.unforgeableNameRng(
+      StandardDeploys.revVault.pk,
+      StandardDeploys.revVault.data.timestamp
+    )
+    // the 11th unforgeable name
+    val unfogeableBytes = (1 to 11).foldLeft(Array.empty[Byte]) {
+      case (_, _) => seedForRevVault.next()
+    }
```

Comment:
I was thinking there is already some kind of support for Iterator and unfold but it seems it's a new stuff in 2.13.
https://github.com/scala/scala/pull/6851

But there is unfold on LazyList. But even that is too much, I found nice `continually` function on Iterator to just drop first 10 elements.
```scala
LazyList.unfold(Array.empty[Byte])(s => (seedForRevVault.next(), s).some).drop(10).head

// Even better without keeping intermediate values
Iterator.continually(seedForRevVault.next()).drop(10).next()
```